bundles/org.eclipse.ui.ide/src/org/eclipse/ui/internal/ide/ChooseWorkspaceDialog.java

```diff
@@ -476,6 +472,26 @@ public void widgetSelected(SelectionEvent e) {
 				}
 			}
 		});
+		parent.addControlListener(new ControlListener() {
+
+			@Override
+			public void controlResized(ControlEvent e) {
+				// browseButton
+				Point size = parent.getSize();
+				if (size.x < 300) {
+					browseButton.setText("..."); //$NON-NLS-1$
```

Comment:
What about using a directory emoji: 📂 ?

Reply:
Good idea. Though, is this supported on all platforms? Is it used elsewhere in eclipse?

Reply:
I found Emojis problematic as support across platform is not guaranteed. I once recommended to have some standard SWT support for a cross platform icon font but it was not getting much attention / acceptance.

Reply:
> I found Emojis problematic as support across platform is not guaranteed

That's probably an outdated impression and an invalid argument. SWT supports emojis very well on all Platform described in https://www.eclipse.org/projects/project-plan.php?planurl=http://www.eclipse.org/eclipse/development/plans/eclipse_project_plan_4_26.xml , and those Platforms do support emojis as well.
If you have some counter-example, please report.

> . I once recommended to have some standard SWT support for a cross platform icon font but it was not getting much attention / acceptance.

It is IMO a bad idea: first the supported OS do already support some cross-platform icons (emojis), and then SWT concept is to integrate in native. Creating another icon palette goes to the opposite goal of what SWT tries to achieve; it would be similar to adding more custom widgets in SWT and rely less and less on native, also emojis have native accessibility capabilities that are much stronger than what SWT would be able to offer by reimplementing some icons; and finally, emojis are cheap and already existing and available and easy t use now, while your suggestion would be more expensive to develop and adopt.